pkg/kinflate/resource/manifestresource.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	manifest "k8s.io/kubectl/pkg/apis/manifest/v1alpha1"
+	interror "k8s.io/kubectl/pkg/kinflate/internal/error"
+	"k8s.io/kubectl/pkg/loader"
+)
+
+// NewFromManifest returns a Resource slice given a manifest object.
+func NewFromManifest(l loader.Loader, m *manifest.Manifest) ([]*Resource, error) {
```

Comment:
I'm wondering if this should be a method of the Application abstraction? 1 method giving you the cooked resources, 1 giving you the raw.

i.e. I'm not sure people should use this function directly, should they? Or maybe you can have this here, and then have `func (a *applicationImpl) Raw() ([]*Resource, error) { return NewFromManifest(a.loader, a.manifest) }`

Reply:
This idea is in alignment with what we implemented in PR #322. We have two functions for application interface. `func (a *applicationImpl) RawResources() (ResourceCollection, error)`  and `func (a *applicationImpl) Resources() (ResourceCollection, error)`. Both `RawResources` and `Resources` have recursive structure. 

Here `NewFromManifest` deals with only one manifest object. It doesn't have any recursion.

Reply:
I think it makes sense, but it's still confusing. I would expect that "NewFromManifest" function, that loads resources from the manifest should load all the resources from the manifest? This is kind of triaging in a confusing way?

Reply:
> I'm wondering if this should be a method of the Application abstraction? 

No, I think it is too concrete to be in the application abstraction.

> i.e. I'm not sure people should use this function directly, should they?

Not sure. But we expose `NewFromPath`, `NewFromConfigMap` and `NewFromSecretGenerator`

The action reasonable to me is that move this to the application pkg and make it private.
Application ~= Manifest

Reply:
I see. It is the function name that causes confusion. Do you have some other name in your mind we can replace `NewFromManifest`?

Reply:
Or maybe we should remove all the references to the `v1alpha1`/Manifest type in Resources, and keep these hidden behind the application?

Reply:
> The action reasonable to me is that move this to the application pkg and make it private.
Application ~= Manifest

Yeah. Or a mix of both:

> we should remove all the references to the v1alpha1/Manifest type in Resources, and keep these hidden behind the application?

As in, we still have things to create resources, but not directly from the Manifest types/sub-types

Reply:
I think this should be implemented as a private method at the ApplicationImpl struct.